### PR TITLE
Fix some issues found by sonar

### PR DESCRIPTION
**Pull request by kangkaisen**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] others

Screen Shot 2022-03-24 at 7 19 08 PM

Screen Shot 2022-03-24 at 7 36 50 PM

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/catalog/Partition.java [hunk 1]
```diff
@@ -465,7 +465,7 @@ public boolean equals(Object obj) {
 
         return (visibleVersion == partition.visibleVersion)
                 && (baseIndex.equals(partition.baseIndex)
-                && distributionInfo.eqauls(partition.distributionInfo));
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/catalog/Partition.java:468]: ???

**Review by murphyatwork — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 3 / 12 (25.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Partition.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/load/LoadJob.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/catalog/HashDistributionInfo.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/catalog/RandomDistributionInfo.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/catalog/Table.java|3|4|75.00%|

File: fe/fe-core/src/main/java/com/starrocks/catalog/Partition.java (hunk 1 above)
Comment (kangkaisen) [reply fe/fe-core/src/main/java/com/starrocks/catalog/Partition.java:468]: Which is amazing!

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 3 / 11 (27.27%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Partition.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/catalog/HashDistributionInfo.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/catalog/RandomDistributionInfo.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/catalog/Table.java|3|4|75.00%|

**Review by satanson — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Pull request merged**